tests/pytests/unit/grains/test_core.py

```diff
@@ -3623,3 +3871,1154 @@ def _mock_is_file(filename):
+        assert "master" in ret
+        assert ret["master"] == "test_master_id"
+
+
+def test__selinux():
+    """
+    test _selinux
+    """
+    with patch.dict(
+        core.__salt__,
+        {
+            "cmd.run": MagicMock(return_value="Enforcing"),
+            "cmd.retcode": MagicMock(return_value=1),
+        },
+    ), patch.object(core, "_linux_bin_exists", MagicMock(return_value=False)):
+        ret = core._selinux()
+        assert ret == {"enabled": False}
+
+    with patch.dict(
+        core.__salt__,
+        {
+            "cmd.run": MagicMock(return_value="Enforcing"),
+            "cmd.retcode": MagicMock(return_value=0),
+        },
+    ), patch.object(core, "_linux_bin_exists", MagicMock(return_value=True)):
+        ret = core._selinux()
+        assert ret == {"enabled": True, "enforced": "Enforcing"}
+
+    with patch.dict(
+        core.__salt__,
+        {
+            "cmd.run": MagicMock(return_value="Disabled"),
+            "cmd.retcode": MagicMock(return_value=0),
+        },
+    ), patch.object(core, "_linux_bin_exists", MagicMock(return_value=True)):
+        ret = core._selinux()
+        assert ret == {"enabled": True, "enforced": "Disabled"}
+
+
+def test__systemd():
+    """
+    test _systemd
+    """
+    with patch.dict(
+        core.__salt__,
+        {
+            "cmd.run": MagicMock(
+                return_value="systemd 254 (254.3-1)\n+PAM +AUDIT -SELINUX -APPARMOR -IMA +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBFDISK +PCRE2 -PWQUALITY +P11KIT -QRENCODE +TPM2 +BZIP2 +LZ4 +XZ +ZLIB +ZSTD +BPF_FRAMEWORK +XKBCOMMON +UTMP -SYSVINIT default-hierarchy=unified"
+            ),
+        },
+    ):
+        ret = core._systemd()
+        assert "version" in ret
+        assert "features" in ret
+        assert ret["version"] == "254"
+        assert (
+            ret["features"]
+            == "+PAM +AUDIT -SELINUX -APPARMOR -IMA +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBFDISK +PCRE2 -PWQUALITY +P11KIT -QRENCODE +TPM2 +BZIP2 +LZ4 +XZ +ZLIB +ZSTD +BPF_FRAMEWORK +XKBCOMMON +UTMP -SYSVINIT default-hierarchy=unified"
+        )
+
+
+def test__clean_value(caplog):
+    """
+    test _clean_value
+    """
+    ret = core._clean_value("key", None)
+    assert not ret
+
+    ret = core._clean_value("uuid", "49e40e2a-63b4-11ee-8c99-0242ac120002")
+    assert ret == "49e40e2a-63b4-11ee-8c99-0242ac120002"
+
+    with patch.object(uuid, "UUID", MagicMock()) as mock_uuid:
+        with caplog.at_level(logging.TRACE):
+            mock_uuid.side_effect = ValueError()
+            ret = core._clean_value("uuid", "49e40e2a-63b4-11ee-8c99-0242ac120002")
+            assert not ret
+            assert (
+                "HW uuid value 49e40e2a-63b4-11ee-8c99-0242ac120002 is an invalid UUID"
+                in caplog.messages
+            )
+
+    values = [
```

Comment:
I'd prefer this to be parametrizing the test, but that would also mean splitting up the UUID part above into a separate test.

I'll let you decide on this one(ie, not a hard change request from me)